internal/broadcaster/utxo_creator_test.go

```diff
@@ -1,83 +1,110 @@
 )
 
 func TestUTXOCreator(t *testing.T) {
-	tt := []struct {
-		name               string
-		outputs            int
-		satoshisPerOutput  uint64
-		startFunc          func(outputs int, satoshisPerOutput uint64) error
-		expectedError      error
-		expectedStartCalls int
+	ks, err := keyset.New()
+	require.NoError(t, err)
+
+	logger := slog.New(slog.NewTextHandler(os.Stdout, &slog.HandlerOptions{Level: slog.LevelDebug}))
+
+	tests := []struct {
+		name                   string
+		getUTXOsResp           sdkTx.UTXOs
+		getBalanceFunc         func(ctx context.Context, address string, constantBackoff time.Duration, retries uint64) (int64, int64, error)
+		expectedBroadcastCalls int
+		expectedError          error
+		requestedUTXOs         int
+		requestedAmountPerUTXO uint64
 	}{
 		{
-			name:              "successful start",
-			outputs:           5,
-			satoshisPerOutput: 1000,
-			startFunc: func(outputs int, satoshisPerOutput uint64) error {
-				return nil
-			},
-			expectedError:      nil,
-			expectedStartCalls: 1,
-		},
-		{
-			name:              "error on start",
-			outputs:           5,
-			satoshisPerOutput: 1000,
-			startFunc: func(outputs int, satoshisPerOutput uint64) error {
-				return errors.New("failed to start UTXO creation")
-			},
-			expectedError:      errors.New("failed to start UTXO creation"),
-			expectedStartCalls: 1,
-		},
-		{
-			name:              "invalid input - zero outputs",
-			outputs:           0,
-			satoshisPerOutput: 1000,
-			startFunc: func(outputs int, satoshisPerOutput uint64) error {
-				return nil // Simulate success to test input handling
+			name:         "success - creates correct UTXOs",
+			getUTXOsResp: sdkTx.UTXOs{{TxID: testdata.TX1Hash[:], Vout: 0, LockingScript: ks.Script, Satoshis: 401}},
+			getBalanceFunc: func(ctx context.Context, address string, constantBackoff time.Duration, retries uint64) (int64, int64, error) {
+				return 400, 0, nil
 			},
-			expectedError:      nil, // Start should handle zero outputs gracefully
-			expectedStartCalls: 1,
+			expectedBroadcastCalls: 1,
+			expectedError:          nil,
+			requestedUTXOs:         4,
+			requestedAmountPerUTXO: 100,
 		},
 		{
-			name:              "invalid input - zero satoshis per output",
-			outputs:           5,
-			satoshisPerOutput: 0,
-			startFunc: func(outputs int, satoshisPerOutput uint64) error {
-				return nil // Simulate success to test input handling
+			name:         "Insufficient balance",
+			getUTXOsResp: sdkTx.UTXOs{{TxID: testdata.TX1Hash[:], Vout: 0, LockingScript: ks.Script, Satoshis: 50}},
+			getBalanceFunc: func(ctx context.Context, address string, constantBackoff time.Duration, retries uint64) (int64, int64, error) {
+				return 100, 0, nil
 			},
-			expectedError:      nil, // Start should handle zero satoshis gracefully
-			expectedStartCalls: 1,
+			expectedBroadcastCalls: 0,
+			expectedError:          errors.New("requested total of satoshis exceeds balance\nrequested: 400, balance: 100"),
```

Comment:
Here the expected error is not a new error but `broadcaster.ErrRequestedSatoshisTooHigh`